ark/utils/spatial_analysis_utils.py

```diff
@@ -12,64 +13,69 @@
 from tqdm.notebook import tqdm
 
 import ark.settings as settings
-from ark.utils import io_utils, misc_utils
+from ark.utils import io_utils, load_utils, misc_utils
 from ark.utils._bootstrapping import compute_close_num_rand
 
 
-def calc_dist_matrix(label_maps, save_path=None):
-    """Generate matrix of distances between center of pairs of cells
+def calc_dist_matrix(label_dir, save_path, prefix='_feature_0'):
+    """Generate matrix of distances between center of pairs of cells.
+
+    Saves each one individually to `save_path`.
 
     Args:
-        label_maps (xarray.DataArray):
-            array of segmentation masks indexed by (fov, cell_id, cell_id, segmentation_label)
+        label_dir (str):
+            path to segmentation masks indexed by `(fov, cell_id, cell_id, segmentation_label)`
         save_path (str):
-            path to save file. If None, then will directly return
-    Returns:
-        dict:
-            Contains a cells x cells matrix with the euclidian
-            distance between centers of corresponding cells for every fov,
-            note that each distance matrix is of type xarray
+            path to save the distance matrices
+        prefix (str):
+            the prefix used to identify label map files in `label_dir`
     """
 
-    # Check that file path exists, if given
-    if save_path is not None:
-        io_utils.validate_paths(save_path, data_prefix=False)
+    # check that both label_dir and save_path exist
+    io_utils.validate_paths([label_dir, save_path])
+
+    # load all the file names in label_dir
+    fov_files = io_utils.list_files(label_dir, substrs=prefix + '.tiff')
 
-    dist_mats_list = []
+    # iterate for each fov
+    for fov_file in fov_files:
+        # retrieve the fov name
+        fov_name = fov_file.replace(prefix + '.tiff', '')
 
-    # Extract list of fovs
-    fovs = label_maps.coords['fovs'].values
+        # load in the data
+        fov_data = load_utils.load_imgs_from_dir(
+            label_dir, [fov_file], match_substring='_feature_0',
+            trim_suffix='_feature_0', xr_channel_names=['segmentation_label']
+        )
+
```

Comment:
Is there a reason you're hard-coding in '_feature_0' in the `load_imgs_from_dir` function, even though `calc_dist_matrix` has a 'prefix' argument?

Reply:
Good point, I'll replace this with `prefix`.